### PR TITLE
Fix issues in quadlet generation

### PR DESCRIPTION
**Pull request by rhatdan**
No reason to volume mount ramalama into container.

### CODE REVIEW

**Comment by rhatdan** [comment]
@ygalblum PTAL

File: ramalama/model.py [hunk 1]
```diff
@@ -128,14 +128,10 @@ def quadlet(self, args, exec_args):
 [Container]
 Device=+/dev/dri
```
Comment (ericcurtin) [inline ramalama/model.py:129]: We should only include these if they exist on the system I think:

/dev/dri
/dev/kfd

not sure if the container will start if they are specified and they are not present.

File: ramalama/model.py [hunk 2]
```diff
@@ -128,14 +128,10 @@ def quadlet(self, args, exec_args):
...
 Device=+/dev/kfd
-Environment=RAMALAMA_TRANSPORT=%s
 Exec=%s
 Image=%s
-Label=RAMALAMA container
-%s
 SecurityLabelDisable=true
-Volume=%s:/usr/bin/ramalama/ramalama:ro
-Volume=%s:/var/lib/ramalama:ro
```
Comment (ericcurtin) [inline ramalama/model.py:138]: Without bind mounting /var/lib/ramalama the container won't have access to models, in the rootless case we do something like:

```
-v/var/home/ecurtin/.local/share/ramalama:/var/lib/ramalama
```

Comment (rhatdan) [reply ramalama/model.py:138]: Good point.  We should just volume mount in the image

**Review by ericcurtin — APPROVED** [review]
I'm not sure about the code here, but assuming these quadlets were generated and tested with a couple of curl commands to show the model is responsive. That's good with me

File: docs/ramalama-serve.1.md [hunk 3]
```diff
@@ -48,7 +48,7 @@ CONTAINER ID  IMAGE                             COMMAND               CREATED
 
 Generate a quadlet for running the AI Model service
 ```
-$ ramalama serve --generate=quadlet granite
+$ ramalama serve --name MyModel --generate=quadlet granite
```
Comment (ygalblum) [inline docs/ramalama-serve.1.md:51]: A few questions:

1. Can't the user chose the model? Or is this what `granite` is supposed to mean? If yes, then shouldn't the name fit the one used in the output?
2. Using `MyModel` as the value of the Container name is a bit confusing. I initially thought it was the name of the model

File: ramalama/model.py [hunk 4]
```diff
@@ -106,18 +106,18 @@ def serve(self, args):
...
         if hasattr(args, "port"):
             port_string = f"PublishPort={args.port}"
 
         name_string = ""
-        if hasattr(args, "name") and args.name != "":
-            name_string = f"Name={args.name}"
+        if hasattr(args, "name") and args.name:
+            name_string = f"ContainerName={args.name}"
 
         print(
```
Comment (ygalblum) [inline ramalama/model.py:122]: Not blocking, but have you considered using `f""` instead of `%`?

Comment (rhatdan) [reply ramalama/model.py:122]: I have used both in the code,  Eric tends to use the f"" and I tend to use the %s method. But we should go through the code and use one for consistency.

**Comment by rhatdan** [comment]
@ericcurtin I am not thrilled with what we have here so far, but this should work.  I think we should have an example with OCI Artifact models which would continue to use the Artifact name rather then a path to the model for the generated quadlet.  Perhaps generating two quadlets, a QUADLET.volume and QUADLET.container.

File: ramalama/model.py (hunk 4 above)
Comment (ericcurtin) [reply ramalama/model.py:122]: I understand the desire for printf-style, that's normally what I lean towards in new languages. But I am starting to prefer f"", you don't have to count out the args. This code is a good example, I really have to count the %'s and the args to figure out which is the 5th parameter. Also with f"" one doesn't have to worry about type.

I'm leaning towards f""

These python style {} formatted strings actually made the C++ standard recently via std::format (which was previously https://github.com/fmtlib/fmt). It's actually faster than printf family of functions.

**Comment by ericcurtin** [comment]
Happy to merge on green @rhatdan ... We can always revisit in other PRs anyway :)

I'll be offline shortly

File: ramalama/model.py (hunk 4 above)
Comment (ericcurtin) [reply ramalama/model.py:122]: Genuinely I don't mind using both either where appropriate. flake8 and black provides us some standards, not everything has to be standardised on one solution. Like maybe sometimes one does want to explicitly represent something as a %s and that's fine with me!

**Pull request merged**